src/dropdown/dropdown.ts

```diff
@@ -82,7 +89,8 @@ export class NgbDropdown {
   }
 
   closeFromOutsideClick($event) {
-    if (this.autoClose && $event.button !== 2 && !this._isEventFromToggle($event)) {
+    if (this.autoClose && this._shouldcloseFromInsideClick($event) && $event.button !== 2 &&
```

Comment:
@pkozlowski-opensource Hm, this methods name is a bit misleading since it doesn't take into account if the click is from outside the dropdown. One solution is to implement the `!this.elementRef.nativeElement.contains($event.target)` check in the if statement to exclude inside clicks, thus making this method only responding to outside clicks.

Another solution just to rename it to `closeFromClick`.